akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -398,12 +761,6 @@ abstract class GraphStageLogic private[stream] (inCount: Int, outCount: Int) {
 
   // Internal hooks to avoid reliance on user calling super in preStart
   protected[stream] def beforePreStart(): Unit = {
```

Comment:
ah, since you are only allowing timers from preStart(). This solves one of my issues, as this allows the timer related things to be pulled out to a separate class that is not a subclass.